### PR TITLE
Update positional encoding of modeling_smt.py

### PR DESCRIPTION
**Pull request by eric-ayllon**
Changed the positional encoding 2D to be device agnostic and not rely on samples having a batch dimension.

Changed the positional encoding 1D to be device agnostic, not relying on samples having a batch dimension, and removing two permutations and calls to tensor.contiguous(). Positional encoding 1D now has the following shape: (len_max, dim) instead of (dim, len_max).

The positional encoding tensors of both PositionalEncoding1D and PositionalEncoding2D are now registered as non-persistent tensors (they will not be saved to the model state_dict), allowing torch to move them to the appropriate device along with the whole model.

Appropriate testing is required.

### CODE REVIEW

**Comment by antoniorv6** [comment]
Updates report 2.5 SER in GrandStaff, so pull request is accepted.
